Dockerfile

```diff
@@ -2,20 +2,16 @@ FROM nginx:1.23.3-alpine
 
 LABEL maintainer="fehguy"
 
-ENV BASE_URL ""
+ENV BASE_URL "/" \
+    NGINX_PORT "8080"
 
-COPY nginx.conf /etc/nginx/
+COPY nginx.conf /etc/nginx/templates/default.conf.template
 
 COPY ./index.html /usr/share/nginx/html/
 COPY ./dist/oauth2-redirect.html /usr/share/nginx/html/
 COPY ./dist/* /usr/share/nginx/html/dist/
-COPY ./docker-run.sh /usr/share/nginx/
+COPY ./docker-run.sh /docker-entrypoint.d/91-docker-run.sh
 
-
-RUN chmod +x /usr/share/nginx/docker-run.sh && \
-    chmod -R a+rw /usr/share/nginx && \
-    chmod -R a+rw /etc/nginx
+RUN chmod +x /docker-entrypoint.d/91-docker-run.sh
 
 EXPOSE 8080
```

Comment:
If we allow to change the port via `PORT` env var, how does it relate to `EXPOSE` directive?

Reply:
According to the [docker documentation](https://docs.docker.com/engine/reference/builder/#expose) and personal experience it does not really matter: "The EXPOSE instruction **does not actually publish** the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published. To actually publish the port when running the container, use the `-p` flag on `docker run` to publish and map one or more ports, or the `-P` flag to publish all exposed ports and map them to high-order ports."

Reply:
Sounds good.